### ISSUE TITLE
Obtain WorkChain subclass from a WorkCalculation

### ISSUE THREAD

**Issue by oschuett** [aiidateam/aiida_core#975]
I would like to write a generic routine for resubmitting work calculations. This requires access to the WorkChain subclass. Ideally, I would like to write sth. like this:
```
w = load_node(42)                    # an AiiDA WorkCalculation object
wclass = w.get_workchain_subclass()  # e.g. NanoribbonWorkChain 
winps = w.get_inputs_dict()
submit(wclass, **winps)              # resubmit
```

**Comment by sphuber** [comment]
I am almost done on a fix for this. I will add this in a PR for issue #1064

**Comment by muhrin** [comment]
Fixed by PR #1294
